docs/shell_setup.md

```diff
@@ -0,0 +1,180 @@
+
+=== "zsh"
+    ```bash linenums="1" title="~/.zshrc"
+    alias k=kubectl
+    source <(kubectl completion zsh)
+    complete -F __start_kubectl k
+    ulimit -n 2048          # kubectl opens one cnx (file) per resource
+    ```
+
+=== "bash"
+    ```bash linenums="1" title="~/.bashrc"
+    alias k=kubectl
+    source <(kubectl completion bash)
+    complete -o default -F __start_kubectl k
+    ulimit -n 2048          # kubectl opens one cnx (file) per resource
+    ```
+
+OK This is the basic:
+
+1) instead of typing `kubectl`, just type `k`. It's 6 less characters !
+
+2) add completion to your shell. So you can type `k <tab>` to get help or `k po<tab>` to get completion to `k port-forward`
+
+3) make `k` also use completion
+
+4) increase the number of open files to 2048. This is needed in some cases when you're dealing with a lot of resources and `kubectl` open many connexions against the cluster.
+
+## Cloud commands
+
+### Gcloud
+
+=== "Apple Mac OSX ZSH"
+    ```bash linenums="1" title="~/.zshrc"
+    source "/usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/path.zsh.inc"
+    source "/usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/completion.zsh.inc"
+    ```
+
+=== "Apple Mac OSX BASH"
+    ```bash linenums="1" title="~/.bashrc"
+    source "/usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/path.bash.inc"
+    source "/usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/completion.bash.inc"
+    ```
+
+=== "Linux ZSH"
+    ```bash linenums="1" title="~/.zshrc"
+    # path may vary depending on where you installed gcloud command
+    $HOME/kubernetes/google-cloud-sdk/path.zsh.inc
+    $HOME/kubernetes/google-cloud-sdk/completion.zsh.inc
+    ```
+
+=== "Linux BASH"
+    ```bash linenums="1" title="~/.bashrc"
+    # path may vary depending on where you installed gcloud command
+    $HOME/kubernetes/google-cloud-sdk/path.bash.inc
+    $HOME/kubernetes/google-cloud-sdk/completion.bash.inc
+    ```
+
+### AWS CLI
+
+=== "ZSH"
+    ```bash linenums="1" title="~/.zshrc"
+    complete -C '/usr/local/bin/aws_completer' aws
+    ```
+
+=== "BASH"
+    ```bash linenums="1" title="~/.bashrc"
+    complete -C '/usr/local/bin/aws_completer' aws
+    ```
+
+## ZSH
+
+Mac OSX default shell is now `zsh` so we'll focus on ZSH now on. ZSH is also widely available in Linux and may somtimes be the default too.
+
+!!! note "ZSH deprecation warning"
+    To get rid of the deprecation warning in OSX, export this variable in your shell. Add it to your `.zshrc` file for persistance:
+
+    ```bash linenums="1" title="~/.zshrc"
+    export BASH_SILENCE_DEPRECATION_WARNING=1
+    ```
+### Oh My ZSH !
+
+Oh My Zsh is a delightful, open source, community-driven framework for managing your Zsh configuration. It comes bundled with thousands of helpful functions, helpers, plugins, themes, and a few things that make you shout... `"Oh My ZSH!"`
+
+#### install
```

Comment:
install  -> Install `Oh My ZSH` framework

Reply:
install is a submenu of `Oh My ZSH !` so I guess it's ok ?